src/main/java/be/uantwerpen/scicraft/entity/EntropyCreeperEntity.java

```diff
@@ -196,8 +212,8 @@ private void shuffle() {
 
         // Shuffle blocks
         Scicraft.LOGGER.debug("size: " + blocksToShuffle.size());
-        if (world.getGameRules().getBoolean(GameRules.DO_MOB_GRIEFING)) {
-            for(int i = 0; i <= blocksToShuffle.size() * SHUFFLE_PERCENTAGE; i++){
```

Comment:
Seems like the real problem is over here. The <= has to be a < for the loop to be skipped when it's empty (no explicit check needed). My bad..